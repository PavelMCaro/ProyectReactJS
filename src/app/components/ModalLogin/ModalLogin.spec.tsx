import * as React from 'react';
import * as Enzyme from 'enzyme'
import { ModalLogin } from './index';
import { shallow, mount, } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('Test for Modal login', () => {

    const props = {
        closeModal: () => {},
        emailValue: "andriu.montalvoc@gmail.com",
        passwordValue: "orbisfront123",
        changeEmail: () => {},
        changePassword: () => {},
        blurEmail: () => {},
        blurPassword: () => {},
        typeErrorEmail: 0,
        typeErrorPassword: 0,
        formSubmit: () => {},
        errorLogin: 0,
        textErrorLogin: "Error Login"
    }
    const component = shallow((
        <ModalLogin {...props}
        /> 
    ));

    it('Test Show Component', () =>{
        console.log(component.debug());
        expect(component).toMatchSnapshot();
    });
    it('Test exits Tags', () =>{
        expect(component.find('Modal').length).toEqual(1);
    });
});