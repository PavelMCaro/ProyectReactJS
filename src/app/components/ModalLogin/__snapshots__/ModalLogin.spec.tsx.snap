// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Modal login Test Show Component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    blurEmail={[Function]}
    blurPassword={[Function]}
    changeEmail={[Function]}
    changePassword={[Function]}
    closeModal={[Function]}
    emailValue="andriu.montalvoc@gmail.com"
    errorLogin={0}
    formSubmit={[Function]}
    passwordValue="orbisfront123"
    textErrorLogin="Error Login"
    typeErrorEmail={0}
    typeErrorPassword={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <styled.form
        id="login-form"
        onSubmit={[Function]}
      >
        <styled.img
          onClick={[Function]}
          src="src/app/assets/images/closemodal-icon.png"
        />
        <styled.h1>
          Ingresa aquí
        </styled.h1>
        <styled.input
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Ingresa tu Email"
          showError={false}
          value="andriu.montalvoc@gmail.com"
        />
        <styled.div
          show={false}
        >
          Campo requerido.
        </styled.div>
        <styled.div
          show={false}
        >
          Por favor, introduce una dirección de correo válida.
        </styled.div>
        <styled.input
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Contraseña"
          showError={false}
          type="password"
          value="orbisfront123"
        />
        <styled.div
          show={false}
        >
          Campo requerido.
        </styled.div>
        <styled.div
          show={false}
        >
          Por favor, introduzca al menos 4 caracteres.
        </styled.div>
        <styled.button
          type="submit"
        >
          Iniciar Sesión
        </styled.button>
        <styled.div
          showError={false}
        >
          Error Login
        </styled.div>
      </styled.form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.img
            onClick={[Function]}
            src="src/app/assets/images/closemodal-icon.png"
          />,
          <styled.h1>
            Ingresa aquí
          </styled.h1>,
          <styled.input
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Ingresa tu Email"
            showError={false}
            value="andriu.montalvoc@gmail.com"
          />,
          <styled.div
            show={false}
          >
            Campo requerido.
          </styled.div>,
          <styled.div
            show={false}
          >
            Por favor, introduce una dirección de correo válida.
          </styled.div>,
          <styled.input
            name="password"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Contraseña"
            showError={false}
            type="password"
            value="orbisfront123"
          />,
          <styled.div
            show={false}
          >
            Campo requerido.
          </styled.div>,
          <styled.div
            show={false}
          >
            Por favor, introduzca al menos 4 caracteres.
          </styled.div>,
          <styled.button
            type="submit"
          >
            Iniciar Sesión
          </styled.button>,
          <styled.div
            showError={false}
          >
            Error Login
          </styled.div>,
        ],
        "id": "login-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onClick": [Function],
            "src": "src/app/assets/images/closemodal-icon.png",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Ingresa aquí",
          },
          "ref": null,
          "rendered": "Ingresa aquí",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "name": "email",
            "onBlur": [Function],
            "onChange": [Function],
            "placeholder": "Ingresa tu Email",
            "showError": false,
            "value": "andriu.montalvoc@gmail.com",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Campo requerido.",
            "show": false,
          },
          "ref": null,
          "rendered": "Campo requerido.",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Por favor, introduce una dirección de correo válida.",
            "show": false,
          },
          "ref": null,
          "rendered": "Por favor, introduce una dirección de correo válida.",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "name": "password",
            "onBlur": [Function],
            "onChange": [Function],
            "placeholder": "Contraseña",
            "showError": false,
            "type": "password",
            "value": "orbisfront123",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Campo requerido.",
            "show": false,
          },
          "ref": null,
          "rendered": "Campo requerido.",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Por favor, introduzca al menos 4 caracteres.",
            "show": false,
          },
          "ref": null,
          "rendered": "Por favor, introduzca al menos 4 caracteres.",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Iniciar Sesión",
            "type": "submit",
          },
          "ref": null,
          "rendered": "Iniciar Sesión",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Error Login",
            "showError": false,
          },
          "ref": null,
          "rendered": "Error Login",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <styled.form
          id="login-form"
          onSubmit={[Function]}
        >
          <styled.img
            onClick={[Function]}
            src="src/app/assets/images/closemodal-icon.png"
          />
          <styled.h1>
            Ingresa aquí
          </styled.h1>
          <styled.input
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Ingresa tu Email"
            showError={false}
            value="andriu.montalvoc@gmail.com"
          />
          <styled.div
            show={false}
          >
            Campo requerido.
          </styled.div>
          <styled.div
            show={false}
          >
            Por favor, introduce una dirección de correo válida.
          </styled.div>
          <styled.input
            name="password"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Contraseña"
            showError={false}
            type="password"
            value="orbisfront123"
          />
          <styled.div
            show={false}
          >
            Campo requerido.
          </styled.div>
          <styled.div
            show={false}
          >
            Por favor, introduzca al menos 4 caracteres.
          </styled.div>
          <styled.button
            type="submit"
          >
            Iniciar Sesión
          </styled.button>
          <styled.div
            showError={false}
          >
            Error Login
          </styled.div>
        </styled.form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.img
              onClick={[Function]}
              src="src/app/assets/images/closemodal-icon.png"
            />,
            <styled.h1>
              Ingresa aquí
            </styled.h1>,
            <styled.input
              name="email"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Ingresa tu Email"
              showError={false}
              value="andriu.montalvoc@gmail.com"
            />,
            <styled.div
              show={false}
            >
              Campo requerido.
            </styled.div>,
            <styled.div
              show={false}
            >
              Por favor, introduce una dirección de correo válida.
            </styled.div>,
            <styled.input
              name="password"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Contraseña"
              showError={false}
              type="password"
              value="orbisfront123"
            />,
            <styled.div
              show={false}
            >
              Campo requerido.
            </styled.div>,
            <styled.div
              show={false}
            >
              Por favor, introduzca al menos 4 caracteres.
            </styled.div>,
            <styled.button
              type="submit"
            >
              Iniciar Sesión
            </styled.button>,
            <styled.div
              showError={false}
            >
              Error Login
            </styled.div>,
          ],
          "id": "login-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onClick": [Function],
              "src": "src/app/assets/images/closemodal-icon.png",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Ingresa aquí",
            },
            "ref": null,
            "rendered": "Ingresa aquí",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "email",
              "onBlur": [Function],
              "onChange": [Function],
              "placeholder": "Ingresa tu Email",
              "showError": false,
              "value": "andriu.montalvoc@gmail.com",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Campo requerido.",
              "show": false,
            },
            "ref": null,
            "rendered": "Campo requerido.",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Por favor, introduce una dirección de correo válida.",
              "show": false,
            },
            "ref": null,
            "rendered": "Por favor, introduce una dirección de correo válida.",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "password",
              "onBlur": [Function],
              "onChange": [Function],
              "placeholder": "Contraseña",
              "showError": false,
              "type": "password",
              "value": "orbisfront123",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Campo requerido.",
              "show": false,
            },
            "ref": null,
            "rendered": "Campo requerido.",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Por favor, introduzca al menos 4 caracteres.",
              "show": false,
            },
            "ref": null,
            "rendered": "Por favor, introduzca al menos 4 caracteres.",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Iniciar Sesión",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Iniciar Sesión",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Error Login",
              "showError": false,
            },
            "ref": null,
            "rendered": "Error Login",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
